### PR TITLE
SubGroups page

### LINKED ISSUE #8: Implement Subgroups page in Group detail
* The goal of this task is to implement view of all subgroups on Group detail page.
* The subgroups should be displayed the same way as they are in the `vo detail page`.
* You should use the same components - `GroupsTreeComponent` and `GroupsListComponent`.
* When implementing the `create` button you should use the `CreateGroupDialogComponent`. You also need to edit this component. The component should have two versions. One is for creating a group without a parent group (this is now implemented) and the second version is for creating a subgroup. If the `CreateGroupDialogData` contains a parent group, the second version should be displayed. If there is no parent group in this data, the first version should be displayed. 
* The goal of this task does NOT include implementation of subgroups delete functionality.

### PR DESCRIPTION
**Pull request by DanoFecko**
resolves #8 
I added static properties (Vo, group, member) to Side-menu-service so they dont have to be pulled multiple times from backend, but i am not sure if this is correct solution to this problem.

### CODE REVIEW

File: src/app/shared/side-menu/side-menu-item.service.ts [hunk 1]
```diff
@@ -14,9 +14,15 @@ export class SideMenuItemService {
   constructor(
     private translate: TranslateService,
     private utils: UtilsService
-  ) { }
+  ) {
+  }
+
+  static vo: Vo;
+  static group: Group;
+  static member: RichMember;
```
Comment (Vojtech-Sassmann) [inline src/app/shared/side-menu/side-menu-item.service.ts]: As you thought, this is not a good idea. The purpose of this service is to share the code responsible for creating side menu items. The side menu items have to be downloaded on the pages where they should be displayed.

Comment (Vojtech-Sassmann) [reply src/app/shared/side-menu/side-menu-item.service.ts]: But, you do not have to download the data from backend on all subpages. You can do this in the `group-detail-page` component and use nested routing to display the subpages(overview, subgroups, etc...). This way it is now done in the `vo-detail-page` component. Check out the component and the `vos-routing.module.ts`.

File: src/app/vos/groups-tree/groups-tree.component.ts [hunk 2]
```diff
@@ -57,8 +57,8 @@ export class GroupsTreeComponent implements OnChanges {
     }
 
     idGroupMap.forEach((group: TreeGroup, id: number, map: Map<number, TreeGroup>) => {
-      if (group.parentGroupId != null) {
-        const updatedParentGroup: TreeGroup = map.get(group.parentGroupId);
+      const updatedParentGroup: TreeGroup = map.get(group.parentGroupId);
+      if (updatedParentGroup != null) {
```
Comment (Vojtech-Sassmann) [inline src/app/vos/groups-tree/groups-tree.component.ts]: What is the reason for these changes?

File: src/app/vos/subgroups/subgroups.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import {Group} from '../../core/models/Group';
+import {CreateGroupDialogComponent} from '../../shared/components/dialogs/create-group-dialog/create-group-dialog.component';
+import {GroupSelectChange} from '../groups-list/groups-list.component';
+
+@Component({
+  selector: 'app-subgroups-tab',
+  templateUrl: './subgroups.component.html',
+  styleUrls: ['./subgroups.component.scss']
+})
+export class SubgroupsComponent implements OnInit {
```
Comment (Vojtech-Sassmann) [inline src/app/vos/subgroups/subgroups.component.ts]: Didn't you forget to delete this component?

File: src/app/vos/groups-tree/groups-tree.component.ts (hunk 2 above)
Comment (DanoFecko) [reply src/app/vos/groups-tree/groups-tree.component.ts]: some group might have parent id, but the parent doesn't have to be in map, so if its not it wont throw error when trying to addChild to it.

Comment (Vojtech-Sassmann) [reply src/app/vos/groups-tree/groups-tree.component.ts]: I see. This you have discovered a bug but this is not probably a correct solution. Please, add there a //FIXME comment.

File: src/app/shared/side-menu/side-menu-item.service.ts (hunk 1 above)
Comment (DanoFecko) [reply src/app/shared/side-menu/side-menu-item.service.ts]: ok, so i tried get ids from url through activated router, but it always returned "undefined", i fix it with "manual" parsing of url, but i guess thats not correct either. I would ask you to have look on it when you get back. For now i will push what i have.

File: src/app/vos/subgroups/subgroups.component.ts (hunk 3 above)
Comment (DanoFecko) [reply src/app/vos/subgroups/subgroups.component.ts]: Yes i did

File: src/app/shared/side-menu/side-menu-item.service.ts (hunk 1 above)
Comment (Vojtech-Sassmann) [reply src/app/shared/side-menu/side-menu-item.service.ts]: I think the problem is that you are trying to access parameters of parent router. Chceck how it is done in Vo groups compoment.

File: src/app/vos/group-detail-page/group-subgroups/group-subgroups.component.html [hunk 4]
```diff
@@ -1,3 +1,18 @@
-<p>
-  group-subgroups works!
-</p>
+<button class="ml-3" mat-flat-button color="accent" (click)="onCreateGroup()">
```
Comment (Vojtech-Sassmann) [inline src/app/vos/group-detail-page/group-subgroups/group-subgroups.component.html]: Could you also add a subtitle for this page like it is on other pages, e.g. `vo-groups`?

**Comment by DanoFecko** [comment]
fixed and rebased

**Comment by DanoFecko** [comment]
ok, not fixed, apparently there's been added relevant changes with rebase
